scripts/obtain/install.sh

```diff
@@ -71,43 +40,81 @@ if [ -t 1 ]; then
     fi
 fi
 
-#Standardise OS name to what is put into filenames of the tarballs.
-current_os()
-{
+say_err() {
+    printf "%b\n" "${red:-}dotnet_install: Error: $1${normal:-}" >&2
+}
+
+say() {
+    # using stream 3 (defined in the beginning) to not interfere with stdout of functions
+    # which may be used as return value
+    printf "%b\n" "${cyan:-}dotnet_install:${normal:-} $1" >&3
+}
+
+say_verbose() {
+    if [ "$verbose" = true ]; then
+        say "$1"
+    fi
+}
+
+get_current_os_name() {
+    eval $invocation
+    
     local uname=$(uname)
     if [ "$uname" = "Darwin" ]; then
         echo "osx"
+        return 0
     else
         # Detect Distro
         if [ "$(cat /etc/*-release | grep -cim1 ubuntu)" -eq 1 ]; then
             echo "ubuntu"
+            return 0
         elif [ "$(cat /etc/*-release | grep -cim1 centos)" -eq 1 ]; then
             echo "centos"
+            return 0
         elif [ "$(cat /etc/*-release | grep -cim1 rhel)" -eq 1 ]; then
             echo "rhel.7"
+            return 0
         elif [ "$(cat /etc/*-release | grep -cim1 debian)" -eq 1 ]; then
             echo "debian"
+            return 0
         fi
     fi
+    
+    say_err "OS name could not be detected"
+    return 1
 }
 
 machine_has() {
+    eval $invocation
+    
     type "$1" > /dev/null 2>&1
     return $?
 }
 
-#Not 100% sure at the moment that these checks are enough. We might need to take version into account or do something
-#more complicated. This seemed like a good beginning though as it should catch the default "clean" machine case and give
-#people an appropriate hint.
+check_min_reqs() {
+    if ! machine_has "curl"; then
+        say_err "curl is required to download dotnet. Install curl to proceed."
+        return 1
+    fi
+    
+    return 0
+}
+
 check_pre_reqs() {
-    local os=$(current_os)
-    local _failing=false;
+    eval $invocation
+    
+    local failing=false;
 
-    if [ "$DOTNET_INSTALL_SKIP_PREREQS" = "1" ]; then
+    if [ "${DOTNET_INSTALL_SKIP_PREREQS:-}" = "1" ]; then
         return 0
     fi
 
     if [ "$(uname)" = "Linux" ]; then
+        [ -z "$(ldconfig -p | grep libunwind)" ] && say_err "Unable to locate libunwind. Install libunwind to continue" && _failing=true
```

Comment:
@krwq A few lines below seem to be doing the same thing - https://github.com/dotnet/cli/pull/1648/files#diff-f2668bfadfa3e5acb46a3b133b6ba228R119.